app/reducers/wayEditingHistory.js

```diff
@@ -18,10 +18,30 @@ function wayEditingHistory (state = createDefaultState(), action) {
   switch (action.type) {
     case types.WAY_EDIT_ENTER: {
       const { way } = action
+      let addedNodes
+      let movedNodes
+      let deletedNodes
+      let mergedNodes
+      let modifiedSharedWays
       console.log('way being edited', way)
+      if (action.wayEditingHistory) {
+        // editing a way that's pending upload
+
+        const wayEditingHistory = _cloneDeep(action.wayEditingHistory)
+        addedNodes = wayEditingHistory.addedNodes
```

Comment:
can this be simplified to `{addedNodes, movedNodes, deletedNodes, mergedNodes, modifiedSharedWays} = wayEditingHistory` ?